User submission

clinedav opened the pull request.

fnk0 commented on routes/userSubmissionRoutes.js:
```diff
@@ -0,0 +1,72 @@
+            // Create User Submission object
+            var userSubmission = {
+                username: userName,
+                question: questionId,  // ?????????? Can this be a String
```
It depends. With the current schema it can not. In the current schema we have it setup as reference then this needs to be a Question itself rather than just the ID

fnk0 commented on routes/userSubmissionRoutes.js:
```diff
@@ -0,0 +1,72 @@
+                score: score
+            };
+
+            // Return the user submission object
```
I don't see where you are saving the user submission into the database. Also you need to use the model user submission itself.

So it should be something like:
```
var us = new UserSubmission();
// sets the properties...
yield us.save();
```

clinedav commented:
Okay, I added code to save the user submission to the database.  I wasn't sure how to add a reference to the question so I added the question itself to the user submission object.  Mongo seems to accept it.  Also, I wasn't 100% sure how to add the submission to the database.  I added a method in runnerControler called saveUserSubmssion, based on the save routine in questionController.
